src/cockpit/channels/pcp.py

```diff
@@ -660,6 +660,10 @@ async def run(self, options: JsonObject) -> None:
 
         self.parse_options(options)
         try_import_pcp()
+        # HACK: the mock package test only sets pmapi to true
+        if pmapi is True:  # pragma: no cover
+            raise ChannelError('not-supported')
```

Comment:
I don't understand this -- the "mock" is the little rpm that the integration test creates for on-demand install? If that sets `pmapi`, then shouldn't that count as "supported" (as the package exists), not "not-supported"?

Reply:
So our test installed a fake `cockpit-pcp` package which contained a valid manifest which made the page no longer show "package missing" but instead shows "no metrics".

As our mock package installs a pcp module to show handle "no package" situation, it would then crash and require an `allow_journal_message` for the exception. Instead of allowing that exception, I chose this solution